kickstarter/android-oss
# PAY-1899: Save payment method

# 📲 What

- Save payment methods using PaymentSheet


# 🛠 How
- `createPaymentSource` now accepts a new parameter `intentClientSecret`
- new input added to the viewModel `savePaymentOption`, gets called once the PaymentSheet validated successfully the inputs introduced by the user.
- refreshed the stored cards
- error states using bottom snackbar to match the success messages 

# 👀 See

https://user-images.githubusercontent.com/4083656/190216577-9f26d4b8-6c32-4b50-a539-3aa1a89fb094.mp4



|  |  |

# 📋 QA

- Add a new paymentMethod
- Delete a payment method
- Turn on airplane mode at some point during the process to check error states
- Leave payments screen and come back, all new added payment methods should be there
- Pledge to a project using a payment method added on user settings

# Story 📖

[PAY-1899](https://kickstarter.atlassian.net/browse/PAY-1899)
